lib/ipc/nifm.c

```diff
@@ -0,0 +1,50 @@
+#include<libtransistor/types.h>
+#include<libtransistor/ipc.h>
+#include<libtransistor/ipc/nifm.h>
+#include<libtransistor/ipc/sm.h>
+
+static ipc_object_t nifm_object;
+
+ipc_object_t nifm_g_object;
+
+result_t nifm_init() {
+	result_t r;
+	ipc_request_t rq = ipc_default_request;
+	ipc_response_fmt_t rs = ipc_default_response_fmt;
+
+
+	ipc_object_t nifm_s_object;
+	r = sm_get_service(&nifm_s_object, "nifm:s");
+	if(r)
+		return r;
+
+	rq.request_id = 4;
+	rs.num_move_handles = 1;
+	rs.move_handles = &(nifm_g_object.session);
+	nifm_g_object.object_id = -1;
```

Comment:
use `rs.num_objects` and `rs.objects` (they will still work for non-domains and will properly initialize your `ipc_object_t`)